packages/block-library/src/pullquote/deprecated.js

```diff
@@ -57,7 +58,128 @@ function parseBorderColor( styleString ) {
 	}
 }
 
+// TODO: this is ripe for a bit of a clean up according to the example in https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/#example
 const deprecated = [
+	{
+		attributes: {
+			...blockAttributes,
+		},
+		save( { attributes } ) {
+			const {
+				mainColor,
+				customMainColor,
+				customTextColor,
+				textColor,
+				value,
+				citation,
+				className,
+			} = attributes;
+
+			const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+
+			let figureClasses, figureStyles;
+
+			// Is solid color style
+			if ( isSolidColorStyle ) {
+				const backgroundClass = getColorClassName(
+					'background-color',
+					mainColor
+				);
+
+				figureClasses = classnames( {
+					'has-background': backgroundClass || customMainColor,
+					[ backgroundClass ]: backgroundClass,
+				} );
+
+				figureStyles = {
+					backgroundColor: backgroundClass
+						? undefined
+						: customMainColor,
+				};
+				// Is normal style and a custom color is being used ( we can set a style directly with its value)
+			} else if ( customMainColor ) {
+				figureStyles = {
+					borderColor: customMainColor,
+				};
+			}
+
+			const blockquoteTextColorClass = getColorClassName(
+				'color',
+				textColor
+			);
+			const blockquoteClasses =
+				( textColor || customTextColor ) &&
+				classnames( 'has-text-color', {
+					[ blockquoteTextColorClass ]: blockquoteTextColorClass,
+				} );
+
+			const blockquoteStyles = blockquoteTextColorClass
+				? undefined
+				: { color: customTextColor };
+
+			return (
+				<figure
+					{ ...useBlockProps.save( {
+						className: figureClasses,
+						style: figureStyles,
+					} ) }
+				>
+					<blockquote
+						className={ blockquoteClasses }
+						style={ blockquoteStyles }
+					>
+						<RichText.Content value={ value } multiline />
+						{ ! RichText.isEmpty( citation ) && (
+							<RichText.Content
+								tagName="cite"
+								value={ citation }
+							/>
+						) }
+					</blockquote>
+				</figure>
+			);
+		},
+		migrate( {
+			className,
+			mainColor,
+			customMainColor,
+			customTextColor,
+			...attributes
+		} ) {
+			const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+
+			const style = {};
+			// Block supports: Set style.border.color if a deprecated block has a default style and a `customMainColor` attribute.
+			if ( ! isSolidColorStyle && customMainColor ) {
+				style.border = {
+					color: customMainColor,
+				};
+			}
+
+			// Block supports: Set style.color.background if a deprecated block has a solid style and a `customMainColor` attribute.
+			if ( isSolidColorStyle && customMainColor ) {
+				style.color = {
+					background: customMainColor,
+				};
+			}
```

Comment:
Maybe turn this into a nested

```suggestion
			if ( customMainColor ) {
				if ( ! isSolidColorStyle ) {
					// Block supports: Set style.border.color if a deprecated block has a default style and a `customMainColor` attribute.
					style.border = {
						color: customMainColor,
					};
				} else {
					// Block supports: Set style.color.background if a deprecated block has a solid style and a `customMainColor` attribute.
					style.color = {
						background: customMainColor,
					};
				}
			}
```

to make the logical branching more explicit?